components/servlet-mgt/org.wso2.carbon.identity.servlet.mgt/src/main/java/org/wso2/carbon/identity/servlet/mgt/CustomCXFNonSpringJaxrsServlet.java

```diff
@@ -0,0 +1,713 @@
+               }
+           }
+       }
+
+       if (properties != null && !properties.isEmpty()) {
+           bean.getProperties(true).putAll(properties);
+       }
+   }
+
+   protected void setAllInterceptors(JAXRSServerFactoryBean bean, ServletConfig servletConfig,
+                                     String splitChar) throws ServletException {
+
+       this.setInterceptors(bean, servletConfig, OUT_INTERCEPTORS_PARAM, splitChar);
+       this.setInterceptors(bean, servletConfig, OUT_FAULT_INTERCEPTORS_PARAM, splitChar);
+       this.setInterceptors(bean, servletConfig, IN_INTERCEPTORS_PARAM, splitChar);
+   }
+
+   protected void setSchemasLocations(JAXRSServerFactoryBean bean, ServletConfig servletConfig) {
+
+       String schemas = servletConfig.getInitParameter(SCHEMAS_PARAM);
+       if (schemas != null) {
+           String[] locations = schemas.split(" ");
+           List<String> list = new ArrayList();
+           String[] locationsArray = locations;
+           int locationsLength = locations.length;
+
+           for (int iterator = 0; iterator < locationsLength; ++iterator) {
+               String loc = locationsArray[iterator];
+               String theLoc = loc.trim();
+               if (!theLoc.isEmpty()) {
+                   list.add(theLoc);
+               }
+           }
+
+           if (!list.isEmpty()) {
+               bean.setSchemaLocations(list);
+           }
+       }
+   }
+
+   protected void setDocLocation(JAXRSServerFactoryBean bean, ServletConfig servletConfig) {
+
+       String wadlLoc = servletConfig.getInitParameter(DOC_LOCATION_PARAM);
+       if (wadlLoc != null) {
+           bean.setDocLocation(wadlLoc);
+       }
+   }
+
+   protected void setInterceptors(JAXRSServerFactoryBean bean, ServletConfig servletConfig, String paramName,
+                                  String splitChar) throws ServletException {
+
+       String value = servletConfig.getInitParameter(paramName);
+       if (value != null) {
+           String[] values = value.split(splitChar);
+           List<Interceptor<? extends Message>> list = new ArrayList();
+           String[] valuesArray = values;
+           int valuesLength = values.length;
+
+           for (int iterator = 0; iterator < valuesLength; ++iterator) {
+               String interceptorVal = valuesArray[iterator];
+               Map<String, List<String>> props = new HashMap();
+               String theValue = this.getClassNameAndProperties(interceptorVal, props);
+               if (!theValue.isEmpty()) {
+                   try {
+                       Class<?> intClass = this.loadClass(theValue, "Interceptor");
+                       Object object = intClass.getDeclaredConstructor().newInstance();
+                       this.injectProperties(object, props);
+                       list.add((Interceptor) object);
+                   } catch (ServletException exception) {
+                       throw exception;
```

Comment:
why are we catching this exception? Since the method is already throwing the exception we don't need to catch it

Reply:
Check other places too